e2etests/test/empty_messages.js

```diff
@@ -0,0 +1,210 @@
+      createDeviceMethod(function (err, testDeviceInfo) {
+        provisionedDevice = testDeviceInfo;
+        beforeCallback(err);
+      });
+    });
+
+    after(function (afterCallback) {
+      DeviceIdentityHelper.deleteDevice(provisionedDevice.deviceId, afterCallback);
+    });
+
+    beforeEach(function () {
+      ehClient = eventHubClient.fromConnectionString(hubConnectionString);
+      deviceClient = createDeviceClient(deviceTransport, provisionedDevice);
+      ehReceivers = [];
+    });
+
+    afterEach(function (done) {
+      closeDeviceEventHubClients(deviceClient, ehClient, ehReceivers, done);
+    });
+    it('Device sends a message of zero size and it is received by the service', function (done) {
+      var deviceClientParticipant = 'deviceClient';
+      var ehClientParticipant = 'ehClient';
+      var testRendezvous = new Rendezvous(done);
+      this.timeout(120000);
+      //
+      // The size of non payload is calculated based on the value of the messageId value size
+      // and then on the application property value size plus one for the application property name.
+      //
+      var uuidData = uuid.v4();
+      var message = new Message('');
+      message.messageId = uuidData;
+      ehClient.open()
+        .then(function() {testRendezvous.imIn(ehClientParticipant);})
+        .then(ehClient.getPartitionIds.bind(ehClient))
+        .then(function (partitionIds) {
+          return partitionIds.map(function (partitionId) {
+            return ehClient.createReceiver('$Default', partitionId,{ 'startAfterTime' : Date.now()}).then(function(receiver) {
+              ehReceivers.push(receiver);
+              receiver.on('errorReceived', function(err) {
+                done(err);
+              });
+              receiver.on('message', function (eventData) {
+                  if ((eventData.annotations['iothub-connection-device-id'] === provisionedDevice.deviceId) &&
+                      (eventData.systemProperties.messageId === uuidData) &&
+                      (!eventData.body || (eventData.body.length === 0))) {
+                    receiver.removeAllListeners();
+                    testRendezvous.imDone(ehClientParticipant);
+                  } else {
+                    debug('Incoming device id is: '+eventData.systemProperties['iothub-connection-device-id']);
+                  }
+                });
+            });
+          });
+        })
+        .then(function () {
+          deviceClient.open(function (openErr) {
+            if (openErr) {
+              done(openErr);
+            } else {
+              testRendezvous.imIn(deviceClientParticipant);
+              //need to add properties test when the event hubs node sdk supports it.
+              deviceClient.sendEvent(message, function (sendErr) {
+                if (sendErr) {
+                  return done(sendErr);
+                }
+                testRendezvous.imDone(deviceClientParticipant);
```

Comment:
>testRendezvous.imDone(deviceClientParticipant); [](start = 16, length = 47)

does that need to be in an else clause? in case of error are you going to call done twice I think.

Reply:
I got rid of the return and added the else.

---
In reply to: [227566428](https://github.com/Azure/azure-iot-sdk-node/pull/401#discussion_r227566428) [](ancestors = 227566428)